### DISCUSSION TITLE
Delete device once detected via onboarding

### DISCUSSION THREAD

**Discussion by misch42** [netbox-community/netbox#13069] [General]
Hi,

I have a devcie a long time ago created by onboarding. After that I did an upgrade of netbox and onboarding was not possible and more. Now I want to delete the device. netbox throws the following error:

<class 'django.db.utils.IntegrityError'>

update or delete on table "dcim_device" violates foreign key constraint "netbox_onboarding_onboardingdevice_device_id_cee3ef7b_fk" on table "netbox_onboarding_onboardingdevice"
DETAIL:  Key (id)=(86) is still referenced from table "netbox_onboarding_onboardingdevice".

Python version: 3.9.13
NetBox version: 3.5.1

How can I delete that device?

**Comment by mtinberg** [comment]
Just because you didn't install or upgrade the plugin doesn't mean that the underlying database tables it created were removed.  I don't know if there is an automated way to remove these tables, or if you just need to go into psql and drop the tables for the plugin you don't use anymore, they should be easy enough to find, either by listing in psql and/or reading the model.py of the plugin where they are described.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Michael Schwartzkopff ***@***.***>
Sent: Sunday, July 2, 2023 10:38 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Delete device once detected via onboarding (Discussion #13069)


Hi,

I have a devcie a long time ago created by onboarding. After that I did an upgrade of netbox and onboarding was not possible and more. Now I want to delete the device. netbox throws the following error:

<class 'django.db.utils.IntegrityError'>

update or delete on table "dcim_device" violates foreign key constraint "netbox_onboarding_onboardingdevice_device_id_cee3ef7b_fk" on table "netbox_onboarding_onboardingdevice"
DETAIL: Key (id)=(86) is still referenced from table "netbox_onboarding_onboardingdevice".

Python version: 3.9.13
NetBox version: 3.5.1

How can I delete that device?

**Comment by misch42** [comment]
Yeah, that is what I thought. Will check if this is possible. Thanks.
